modules/platforms/dotnet/Apache.Ignite/Internal/ClientFailoverSocket.cs

```diff
@@ -252,5 +268,95 @@ private IEnumerable<IPAddress> GetIps(string host, bool suppressExceptions = fal
                 throw;
             }
         }
+
+        /// <summary>
+        /// Gets a value indicating whether a failed operation should be retried.
+        /// </summary>
+        /// <param name="exception">Exception that caused the operation to fail.</param>
+        /// <param name="op">Operation code.</param>
+        /// <param name="attempt">Current attempt.</param>
+        /// <returns>
+        /// <c>true</c> if the operation should be retried on another connection, <c>false</c> otherwise.
+        /// </returns>
+        private bool ShouldRetry(Exception exception, ClientOp op, int attempt)
+        {
+            var e = exception;
+
+            while (e != null && !(e is SocketException))
+            {
+                e = e.InnerException;
+            }
+
+            if (e == null)
+            {
+                // Only retry socket exceptions.
```

Comment:
Why only socket exceptions?

Reply:
The idea behind the IEP is to retry errors caused by connection issues. `SocketException` is what you get in .NET in those cases.

Reply:
Got it.